source/CMakeLists.txt

```diff
@@ -19,19 +28,27 @@ set(SRCS
 
 find_package (Threads REQUIRED) #threading
 
-#-------------------------------------------------------------------------------
-#exampleApp executable setup
-if(WIN32) #windows icon resource file, provided from /packaging directory
-    set(WINDOWS_ICON_RESOURCE "${PROJECT_BINARY_DIR}/exampleApp.icon.rc")
-endif()
-
-add_executable(exampleApp ${SRCS} ${WINDOWS_ICON_RESOURCE})
+add_executable(exampleApp MACOSX_BUNDLE ${SRCS} ${ICON_RESOURCE})
 target_link_libraries (exampleApp ${CMAKE_THREAD_LIBS_INIT}) #threading
+
 if(NOT MSVC)
     target_compile_options(exampleApp PRIVATE -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wunused -pedantic) # extra warnings
 endif()
+
+if(APPLE)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_GUI_IDENTIFIER "com.ea.exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_ICON_FILE exampleApp.icns)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_SOURCE_DIR}/packaging/MacOSXBundleInfo.plist.in)
+    set_source_files_properties(${CMAKE_SOURCE_DIR}/packaging/exampleApp.icns PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+endif()
+
 set_property(TARGET exampleApp PROPERTY CXX_STANDARD 14) # we want C++14
-install(TARGETS exampleApp DESTINATION bin) #copy executable to bin folder on install
+if(APPLE)
+    install(TARGETS exampleApp DESTINATION "/") #copy executable to bin folder on install
```

Comment:
So now, when running `make install` on Mac, it will still copy it into `build/dist/bin/exampleApp` or into `builld/dist/exampleApp`. 

My concern is, that some other targets (e.g. integration tests, gdb) may expect binary being in `bin` subfolder. And I want to be sure, that in Mac OS enviroment binaries are expected to be in `[app_root]/` directory and not in `[app_root]/bin/` otherwise, I would keep the binary in `[app_root]/bin` directory to keep it the same across platforms.